include/re_sa.h

```diff
@@ -59,5 +59,12 @@ bool     sa_is_linklocal(const struct sa *sa);
 bool     sa_is_loopback(const struct sa *sa);
 bool     sa_is_any(const struct sa *sa);
 
+#ifdef HAVE_INET6
+void     sa_set_scopeid(struct sa *sa, int scopeid);
+int      sa_scopeid(const struct sa *sa);
+int      sa_cpy_scopeid(struct sa *sa_dst, const struct sa *sa_src);
+bool     sa_is_ipv6ll(const struct sa *sa);
+#endif
+
```

Comment:
we should not use ifdefs in the public api, because it makes the ABI different depending on the ifdef.

on a side note, we should consider remove ifdef HAVE_INET6 and always have it compiled in.
ipv6 is today pretty much standard everywhere.

Reply:
Will remove this for the API functions. Completely removing the switch should be another PR.